api/base/permissions.py

```diff
@@ -53,6 +54,12 @@ def _get_scopes(self, request, view):
             except AttributeError:
                 raise ImproperlyConfigured('TokenHasScope requires the view to define the '
                                            'required_read_scopes attribute')
+            assert is_iterable_but_not_string(view.required_read_scopes), \
+                'The required_read_scopes must be an iterable of CoreScopes'
+            if view.required_read_scopes and isinstance(view.required_read_scopes[0], tuple):
```

Comment:
One other note: it seems that we're trying to check whether the class definition is valid.

That seems like the kind of safeguard that would only need to happen once, to warn the developer they were making a mistake locally... but this middleware class will run the method on every single request. 

Is that performance overhead really necessary? Can we improve that, eg by checking once when the class is first instantiated?  (I'm not at all certain, but I *think* Django uses singletons where possible)